HDFSStorage does not respect closed status

**Problem description**
HDFSStorage does not make use of the closed status. Method `ensureInitializedAndNotClosed` is never used.

This *must* be invoked when any public method is invoked. 

For example, I keep getting this failure locally every now and then, and I traced it to a closed adapter:

```
(trimmed)
Caused by: java.lang.NullPointerException
	at io.pravega.segmentstore.storage.impl.hdfs.HDFSStorage.findAllRaw(HDFSStorage.java:501)
	at io.pravega.segmentstore.storage.impl.hdfs.HDFSStorage.findStatusForSegment(HDFSStorage.java:543)
	at io.pravega.segmentstore.storage.impl.hdfs.HDFSStorage.seal(HDFSStorage.java:239)
	at io.pravega.segmentstore.storage.rolling.RollingStorage.sealActiveChunk(RollingStorage.java:518)
	at io.pravega.segmentstore.storage.rolling.RollingStorage.seal(RollingStorage.java:335)
	at io.pravega.segmentstore.storage.rolling.RollingStorage.delete(RollingStorage.java:445)
(trimmed)
```

**Problem location**
HDFSStorage.

**Suggestions for an improvement**
Invoke `ensureInitializedAndNotClosed` when calling any public method.

--- a/segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/hdfs/HDFSStorage.java
+++ b/segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/hdfs/HDFSStorage.java
@@ -186,7 +186,6 @@ class HDFSStorage implements SyncStorage {
     }
 
     private boolean isSealed(Path path) throws FileNameFormatException {
-        ensureInitializedAndNotClosed();
         return getEpochFromPath(path) == MAX_EPOCH;
     }
 

